src/main/java/com/devonfw/quarkus/productmanagement/service/v1/ProductRestService.java

```diff
@@ -1,112 +1,126 @@
+import org.eclipse.microprofile.openapi.annotations.media.Schema;
+import org.eclipse.microprofile.openapi.annotations.parameters.Parameter;
+import org.eclipse.microprofile.openapi.annotations.responses.APIResponse;
+import org.eclipse.microprofile.openapi.annotations.responses.APIResponses;
 import org.springframework.data.domain.Page;
 
-import com.devonfw.quarkus.productmanagement.logic.UcFindProduct;
-import com.devonfw.quarkus.productmanagement.logic.UcManageProduct;
-import com.devonfw.quarkus.productmanagement.service.v1.model.NewProductDto;
+import com.devonfw.quarkus.productmanagement.domain.model.ProductEntity;
+import com.devonfw.quarkus.productmanagement.domain.repo.ProductRepository;
+import com.devonfw.quarkus.productmanagement.service.v1.mapper.ProductMapper;
 import com.devonfw.quarkus.productmanagement.service.v1.model.ProductDto;
 import com.devonfw.quarkus.productmanagement.service.v1.model.ProductSearchCriteriaDto;
 
-//In Quarkus all JAX-RS resources are treated as CDI beans
-//default is Singleton scope
-@Path("/products")
-// how we serialize response
+@Path("/product/v1")
 @Produces(MediaType.APPLICATION_JSON)
-// how we deserialize params
 @Consumes(MediaType.APPLICATION_JSON)
 public class ProductRestService {
 
-  // using @Context we can inject contextual info from JAXRS(e.g. http request, current uri info, endpoint info...)
-  @Context
-  UriInfo uriInfo;
-
   @Inject
-  UcFindProduct ucFindProduct;
+  ProductRepository productRepository;
 
   @Inject
-  UcManageProduct ucManageProduct;
-
-  @GET
-  public Page<ProductDto> getAll(@BeanParam ProductSearchCriteriaDto dto) {
-
-    return this.ucFindProduct.findProducts(dto);
-  }
-
-  @GET
-  @Path("criteriaApi")
-  public Page<ProductDto> getAllCriteriaApi(@BeanParam ProductSearchCriteriaDto dto) {
-
-    return this.ucFindProduct.findProductsByCriteriaApi(dto);
-  }
+  ProductMapper productMapper;
 
-  @GET
-  @Path("queryDsl")
-  public Page<ProductDto> getAllQueryDsl(@BeanParam ProductSearchCriteriaDto dto) {
-
-    return this.ucFindProduct.findProductsByQueryDsl(dto);
-  }
+  @Context
+  UriInfo uriInfo;
 
   @GET
-  @Path("query")
-  public Page<ProductDto> getAllQuery(@BeanParam ProductSearchCriteriaDto dto) {
+  public Page<ProductDto> getAllOrderedByTitle() {
 
-    return this.ucFindProduct.findProductsByTitleQuery(dto);
+    Page<ProductEntity> products = this.productRepository.findAllByOrderByTitle();
+    if (products.isEmpty()) {
+      return Page.empty();
+    }
+    return this.productMapper.map(products);
   }
 
-  @GET
-  @Path("nativeQuery")
-  public Page<ProductDto> getAllNativeQuery(@BeanParam ProductSearchCriteriaDto dto) {
+  @APIResponses({
+  @APIResponse(responseCode = "201", description = "OK, New Product created", content = @Content(schema = @Schema(implementation = ProductDto.class))),
+  @APIResponse(responseCode = "400", description = "Client side error, invalid request"),
+  @APIResponse(responseCode = "500") })
```

Comment:
Discussion was to use Servicedocgen instead of swagger annotation, but it is unsure whether servicedocgen generated file has the "TryOut" option, which can be used to try out  with the sample data.